### ISSUE TITLE
fzero fails after workspace()

### ISSUE THREAD

**Issue by Alexander-Barth** [JuliaMath/Roots.jl#63]
The last command of:

```julia
using Roots; fzero(x -> x-1, 1) # returns 1, correct ! :-)
workspace()
using Roots; fzero(x -> x-1, 1)
```

fails with
```
ERROR: MethodError: Cannot `convert` an object of type Roots.DerivativeFree to an object of type LastMain.Roots.CallableFunction
This may have arisen from a call to the constructor LastMain.Roots.CallableFunction(...),
since type constructors fall back to convert methods.
 in #derivative_free_setup#6(::Nullable{Array{Float64,1}}, ::Float64, ::Float64, ::Float64, ::Float64, ::Int64, ::Int64, ::Bool, ::Array{Any,1}, ::Roots.#derivative_free_setup, ::Roots.Order0, ::Roots.DerivativeFree, ::Float64) at /home/abarth/.julia/v0.5/Roots/src/find_zero.jl:148
 in #find_zero#10(::Array{Any,1}, ::Function, ::Roots.Order0, ::Roots.DerivativeFree, ::Float64) at /home/abarth/.julia/v0.5/Roots/src/find_zero.jl:411
 in find_zero(::Roots.Order0, ::Roots.DerivativeFree, ::Float64) at /home/abarth/.julia/v0.5/Roots/src/find_zero.jl:410
 in #find_zero#7(::Array{Any,1}, ::Function, ::Function, ::Float64, ::Roots.Order0) at /home/abarth/.julia/v0.5/Roots/src/find_zero.jl:401
 in find_zero(::Function, ::Float64, ::Roots.Order0) at /home/abarth/.julia/v0.5/Roots/src/find_zero.jl:401
 in #derivative_free#11(::Int64, ::Array{Any,1}, ::Function, ::Function, ::Float64) at /home/abarth/.julia/v0.5/Roots/src/find_zero.jl:441
 in #fzero#66(::Array{Any,1}, ::Function, ::Function, ::Int64) at /home/abarth/.julia/v0.5/Roots/src/Roots.jl:72
 in fzero(::Function, ::Int64) at /home/abarth/.julia/v0.5/Roots/src/Roots.jl:70
```

I am using julia 0.5.1 and Roots 0.3.0. Maybe this is an issue in Julia?

```
Julia Version 0.5.1
Commit 6445c82 (2017-03-05 13:25 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i3-3120M CPU @ 2.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, ivybridge)
```
Any help or insight would be really appreciated. And thanks for this nice package!

**Comment by jverzani** [comment]
It is working on v0.6, but I can confirm it doesn't on v0.5. I'm not sure what the root cause would be, so have no idea where to begin to debug this. Sorry.

**Comment by Alexander-Barth** [comment]
Hi John, thank you very much for testing this in v0.6! It really seems to be an issue in julia v0.5 which is already fixed. You might want to close this issue as it is most likely a bug in julia v0.5.

**Comment by ararslan** [comment]
I believe this is related to the infamous issue 265 in Julia, which was fixed in Julia 0.6. I get the behavior you're describing with many packages in 0.5 and earlier.

**Comment by Alexander-Barth** [comment]
Thanks!
